### PR TITLE
Implement automatic creation of ECR repositories

### PR DESCRIPTION
**Pull request by JamesJHPark**
Description of changes:

- This PR is to implement automatic creation of ECR repositories.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vijtrip2** [comment]
/hold
To verify postsubmit account's IAM permission are up-to-date. Otherwise this change will block CD pipeline.

File: scripts/ensure-ecr-repository.sh
Comment (vijtrip2) [inline scripts/ensure-ecr-repository.sh:1]: rename the file to `public-ecr.sh`

File: scripts/ensure-ecr-repository.sh
Comment (vijtrip2) [inline scripts/ensure-ecr-repository.sh:20]: clever, but we need to verify that this tool is available inside the image that we use to run ACK CD jobs.

File: cd/scripts/release-controller.sh
Comment (vijtrip2) [inline cd/scripts/release-controller.sh:168]: you can remove this duplicate call.

File: cd/scripts/release-controller.sh
Comment (vijtrip2) [inline cd/scripts/release-controller.sh:125]: how about changing this to `ensure_repository "$AWS_SERVICE"`?

and then inside that method you ensure both "$AWS_SERVICE"-controller and "$AWS_SERVICE"-chart repositories.

This way you can make sure both the repositories are present before building controller image.

**Review by vijtrip2 — COMMENTED** [review]
Nice work, James! Left some comments.
And before merging i can help you verify that prow iam account has permissions to list and create repositories. :)

File: scripts/ecr-templates/ecr_chart_template.txt
Comment (RedbackThomson) [inline scripts/ecr-templates/ecr_chart_template.txt:1]: This file should be made into a `.json` rather than a `.txt`

File: scripts/ensure-ecr-repository.sh
Comment (RedbackThomson) [inline scripts/ensure-ecr-repository.sh:10]: I don't think this needs to be a variable. You can probably just hardcode `--region=us-east-1` into all of the `aws ecr-public` commands.

**Review by jaypipes — APPROVED** [review]
Love it. Great work on this, @JamesJHPark :)

**Pull request merged**
